plugins/httpboot/plugin.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2018-present the CoreDHCP Authors. All rights reserved
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package netboot
+
+import (
+	"fmt"
+	"github.com/coredhcp/coredhcp/handler"
+	"github.com/coredhcp/coredhcp/logger"
+	"github.com/coredhcp/coredhcp/plugins"
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv6"
+	"net/url"
+	"strings"
+)
+
+var bootFile4 string
+var bootFile6 string
+
+var log = logger.GetLogger("plugins/httpboot")
+
+var Plugin = plugins.Plugin{
+	Name:   "httpboot",
+	Setup6: setup6,
+	Setup4: setup4,
+}
+
+func parseArgs(args ...string) (*url.URL, error) {
+	if len(args) != 1 {
+		return nil, fmt.Errorf("Exactly one argument must be passed to the httpboot plugin, got %d", len(args))
+	}
+	return url.Parse(args[0])
+}
+
+func setup6(args ...string) (handler.Handler6, error) {
+	u, err := parseArgs(args...)
+	if err != nil {
+		return nil, err
+	}
+	bootFile6 = u.String()
+	log.Printf("loaded httpboot plugin for DHCPv6.")
+	return Handler6, nil
+}
+
+func setup4(args ...string) (handler.Handler4, error) {
+	u, err := parseArgs(args...)
+	if err != nil {
+		return nil, err
+	}
+	bootFile4 = u.String()
+	log.Printf("loaded httpboot plugin for DHCPv4.")
+	return Handler4, nil
+}
+
+func Handler6(req, resp dhcpv6.DHCPv6) (dhcpv6.DHCPv6, bool) {
+	decap, err := req.GetInnerMessage()
+	if err != nil {
+		log.Errorf("Could not decapsulate request: %v", err)
+		return nil, true
+	}
+
+	if decap.GetOneOption(dhcpv6.OptionVendorClass) != nil {
+		vc := decap.GetOneOption(dhcpv6.OptionVendorClass).String()
+		if strings.Contains(vc, "HTTPClient") {
+			bf := &dhcpv6.OptionGeneric{
+				OptionCode: dhcpv6.OptionBootfileURL,
+				OptionData: []byte(bootFile6),
+			}
+			resp.AddOption(bf)
+			vid := &dhcpv6.OptionGeneric{
+				OptionCode: dhcpv6.OptionVendorClass,
+				OptionData: []byte("000000HTTPClient"),
+			}
+			resp.AddOption(vid)
+		}
+	}
+
+	return resp, true
+}
+
+func Handler4(req, resp *dhcpv4.DHCPv4) (*dhcpv4.DHCPv4, bool) {
+	if req.GetOneOption(dhcpv4.OptionClassIdentifier) != nil {
+		vc := req.GetOneOption(dhcpv4.OptionClassIdentifier)
+		if strings.Contains(string(vc), "HTTPClient") {
+			bf := &dhcpv4.Option{
+				Code:  dhcpv4.OptionBootfileName,
+				Value: dhcpv4.String(bootFile4),
+			}
+			resp.Options.Update(*bf)
+			vid := &dhcpv4.Option{
+				Code:  dhcpv4.OptionClassIdentifier,
+				Value: dhcpv4.String("HTTPClient"),
+			}
+			resp.Options.Update(*vid)
+		}
+	}
+	return resp, true
```

Comment:
Please don't break the plugin chain, tricky corner case here:
both pxe/httpboot and file plugin break the chain for IPv4, so network boot fails in the pipeline. The file plugin is not under our control, so let us change that one here.

See also https://github.com/ironcore-dev/FeDHCP/pull/151